tests/unit_tests/client/address_book/storage.cc

```diff
@@ -0,0 +1,196 @@
+            boost::filesystem::remove(temp_file);
+        }
+    }
+  catch (boost::filesystem::filesystem_error& er)
+    {
+      std::cerr << __func__ << ": " << er.what();
+    }
+
+  /// @brief Insert test hosts into an address map
+  void ToAddressMap(Subscription source)
+  {
+    for (const auto& host : m_Hosts)
+      {
+        const std::size_t pos = host.find("=");
+        const std::string name = host.substr(0, pos);
+        const std::string address = host.substr(pos);
+        kovri::core::IdentityEx ident;
+        ident.FromBase64(address);
+        m_Addresses[name] = std::make_pair(ident.GetIdentHash(), source);
+      }
+  }
+
+  /// @brief Create new storage instance
+  /// @notes Called after creating custom data directory
+  std::unique_ptr<kovri::client::AddressBookStorage> GetStorageInstance()
+  {
+    return std::make_unique<kovri::client::AddressBookStorage>();
+  }
+
+
+  std::string const test_prefix{"addr_test_"};  // used for temporary test files
+  kovri::client::AddressBook::AddressMap m_Addresses;
+  std::unique_ptr<kovri::client::AddressBookStorage> m_Storage;
+  boost::filesystem::path temp_path = boost::filesystem::temp_directory_path();
+  std::array<std::string, 2> const m_Hosts{{
+      "kovri.i2p=0UVPqAA4xUSfPYPBca24h8fdokhwcJZ4zWvELv-5OsBYTHKtnLzvK7byXtXT~fOV2pExi8vrkgarGTNDfJbB2KCsdVS3V7qwtTvoCGYyklcDBlJsWMj7H763hEz5rt9SzLkcpwhO3t0Zwe6jXL1UB-QW8KxM30t-ZOfPc6OiJ1QpnE6Bo5OUm6jPurQGXdWCAPio5Z-YnRL46n0IHWOQPYYSStJMYPlPS-S75rMIKbZbEMDraRvSzYAphUaHfvtWr2rCSPkKh3EbrOiBYiAP2oWvAQCsjouPgVF2qwQRnBbiAezHedM2gXzkgIyCV2kGOOcHhiihd~7fWwJOloH-gO78QkmCuY-3kp3633v3MBw7pmABr-XNKWnATZOuf2syWVBZbTnOXsWf41tu6a33HOuNsMxAOUrwbu7QRmT4X8X-EITwNlKN6r1t3uoQ~yZm4RKsJUsBGfVtKl8PBMak3flQAg95oV0OBDGuizIQ9vREOWvPGlQCAXZzEg~cUNbfBQAEAAcAAA==",
+      "monero.i2p=3VzGaQQXwzN1iAwaPI17RK~gUqKqMH6fI2dkkGBwdayAPAdiZMyk1KGoTq~q1~HBraPZnz9mZJlzf6WVGCkUmUV3SBjBEbrdL9ud0fArq3P1~Ui9ViR9B7m5EG8smAnFvKZdqS-cnmHploUfIOefoQe0ecM7YYHErZsn3kL-WtvlfoDiSth-edIBpWxeHfmXSKoHSGSJ2snl5p9hxh30KmKj9AB0d4En-jcD83Ep3jsSvtPoQl7tSsh575~q0JJLsqGqm2sR9w4nZr7O58cg-21A2tlZeldM287uoTMb9eHWnYuozUGzzWOXvqg0UxPQSTfwh7YEhx0aRTXT2OFpr84XPoH2M6xIXfEMkFtJEJ-XlM-ILUZkg3kuBEFN7n4mBK~8L0Ht1QCq8L3~y7YnN61sxC0E9ZdyEOoC~nFJxndri9s9NzgZPo5eo6DsZXweOrTAIVQgKFUozL7WXKMlgqBZ5Nl3ijD6MGCIy0fWYHGLJ4jDBY7wrcfynVXFGm4EBQAEAAcAAA==",
+  }};
+};
+
+BOOST_FIXTURE_TEST_SUITE(AddressBookStorageTests, AddressBookStorageFixture)
+
+BOOST_AUTO_TEST_CASE(ValidSaveSubscription)
+{
+  std::size_t saved_addresses = 0;
+
+  // Check we have a storage instance
+  BOOST_CHECK(m_Storage);
+
+  // Ensure successful address insertion
+  BOOST_CHECK_NO_THROW(ToAddressMap(Subscription::Default));
+
+  // Ensure addresses written to storage
+  BOOST_CHECK_NO_THROW(saved_addresses = m_Storage->Save(m_Addresses));
+
+  // Ensure addresses are saved
+  BOOST_CHECK(saved_addresses && saved_addresses == m_Hosts.size());
+}
+
+BOOST_AUTO_TEST_CASE(InvalidSaveSubscription)
+{
+  std::size_t saved_addresses = 0;
```

Comment:
Rather than set this up as a temporary for every case, consider using within the fixture.

Reply:
Will do.